packages/wallet/frontend/src/pages/grants/[grantId].tsx

```diff
@@ -0,0 +1,135 @@
+import { AppLayout } from '@/components/layouts/AppLayout'
+import { PageHeader } from '@/components/PageHeader'
+import { NextPageWithLayout } from '@/lib/types/app'
+import { Badge, getStatusBadgeIntent } from '@/ui/Badge'
+import type {
+  GetServerSideProps,
+  InferGetServerSidePropsType
+} from 'next/types'
+import { z } from 'zod'
+import { formatDate } from '@/utils/helpers'
+import { Grant, grantsService } from '@/lib/api/grants'
+import { Button } from '@/ui/Button'
+import { useDialog } from '@/lib/hooks/useDialog'
+import { ConfirmationDialog } from '@/components/dialogs/ConfirmationDialog'
+import { SuccessDialog } from '@/components/dialogs/SuccessDialog'
+import { ErrorDialog } from '@/components/dialogs/ErrorDialog'
+import { useRouter } from 'next/router'
+import Image from 'next/image'
+
+type GrantPageProps = InferGetServerSidePropsType<typeof getServerSideProps>
+
+const GrantPage: NextPageWithLayout<GrantPageProps> = ({ grant }) => {
+  const [openDialog, closeDialog] = useDialog()
+  const router = useRouter()
+
+  const handleRevokeConfirmation = async (id: string) => {
+    const response = await grantsService.delete(id)
+    if (response.success) {
+      openDialog(
+        <SuccessDialog onClose={closeDialog} content={response.message} />
+      )
+      router.reload()
```

Comment:
I do not think `reload` is the correct method to use, since this performs a page reload for the current URL. If we delete a grant, we should get a 404 page if we `reload` the page.

Reply:
Well, deletion of a grant is not really a Delete, it is a REVOKE. Meaning we do need a reaload here, and a status change from any status to REVOKED. The Grant will not be deleted, just state update.

Reply:
@Tymmmy as the current implementation on rafiki is deleting the grant I think what Radu is saying make sense
this can be changed in the future though

Reply:
OK, I can do a realod to the grants list instead